packages/app/src/components/search/SearchModal.tsx

```diff
@@ -148,15 +148,18 @@ export const SearchModal = ({ toggleModal }: { toggleModal: () => void }) => {
                   values={async () => {
                     // Return a list of entities which are documented.
                     const { items } = await catalogApi.getEntities({
-                      fields: ['metadata.name'],
+                      fields: ['metadata.name', 'metadata.title'],
                       filter: {
                         'metadata.annotations.backstage.io/techdocs-ref':
                           CATALOG_FILTER_EXISTS,
                       },
                     });
 
-                    const names = items.map(entity => entity.metadata.name);
-                    names.sort();
+                    const names = items.map(entity => ({
+                      value: entity.metadata.name,
+                      label: entity.metadata.title ?? entity.metadata.name,
```

Comment:
Can we use the Presentation API - it was built for handling this type of things - instead? More details here: https://github.com/backstage/backstage/pull/17474. This would go for the `SearchPage` as well 👍

Reply:
I think this could make sense, but it's not straight forward to change this without some extracting of this logic elsewhere.

Using the presentation API for this within this async function could be done in two ways:
 * Use a component instead of string as label (and hoping MUI renders it correctly - its API only supports strings as labels). This would not allow for ordering the names alpha-numerically.
 * Use the presentation API's `forEntity` and wait for each promise to resolve, then take the resulting array (, sort it) and return that.

I have now done the latter - extracted this fetching of items to a separate function (hook), re-used in both the search page and modal.